File: generate_assets.js
const fs = require('fs');
const path = require('path');

const files = {
    "head": "sk-head.png",
    "headJump": "sk_jump.png",
    "wifeHead": "jane.png",
    "smallDog": "sk_dog.png",
    "bigDog": "bigdog.jpg",
    "dogOne": "dogone.jpg",
    "dogTwo": "dogtwo.jpg"
};

let jsContent = "const ASSETS = {\n";

for (const [key, fileName] of Object.entries(files)) {
    const filePath = path.join(__dirname, fileName);
    if (fs.existsSync(filePath)) {
        console.log(`Encoding ${fileName}...`);
        const fileData = fs.readFileSync(filePath);
        const ext = path.extname(fileName).toLowerCase();
        let mimeType = 'image/png';
        if (ext === '.jpg' || ext === '.jpeg') {
            mimeType = 'image/jpeg';
        }
        const b64 = `data:${mimeType};base64,${fileData.toString('base64')}`;
        jsContent += `    ${key}: "${b64}",\n`;
    } else {
        console.warn(`Warning: ${fileName} not found!`);
        // If file not found, we might want to keep the old value if we were parsing assets.js, 
        // but here we are regenerating it. If source is missing, we can't bundle it. 
        // Ideally we should fail or ensure files exist.
    }
}

jsContent += "};\n";

fs.writeFileSync(path.join(__dirname, 'assets.js'), jsContent);
console.log("Done! assets.js created.");

File: generate_assets.py
import base64
import os

def image_to_base64(path):
    mime_type = "image/jpeg" if path.lower().endswith(('.jpg', '.jpeg')) else "image/png"
    with open(path, "rb") as image_file:
        return f"data:{mime_type};base64," + base64.b64encode(image_file.read()).decode('utf-8')

assets = {}
files = {
    "head": "c:/Endless Runner/sk-head.png",
    "headJump": "c:/Endless Runner/sk_jump.png",
    "wifeHead": "c:/Endless Runner/jane.png",
    "smallDog": "c:/Endless Runner/sk_dog.png",
    "bigDog": "c:/Endless Runner/bigdog.jpg"
}

print("Generating assets.js...")
js_content = "const ASSETS = {\n"

for key, path in files.items():
    if os.path.exists(path):
        print(f"Encoding {path}...")
        b64 = image_to_base64(path)
        js_content += f'    {key}: "{b64}",\n'
    else:
        print(f"Warning: {path} not found!")

js_content += "};\n"

with open("c:/Endless Runner/assets.js", "w") as f:
    f.write(js_content)

print("Done! assets.js created.")
